frontend/src/data/parseReportData.ts

```diff
@@ -174,54 +174,73 @@ export const implementationFromData = (
     dialects: data.dialects.map((uri) => Dialect.forURI(uri)),
   }) as Implementation;
 
-export const parseImplementationData = (
-  loaderData: Record<string, ReportData>,
+/**
+ * Computes a summarized implementation report for the
+ * passed implementation using all the dialect reports data
+ * that was fetched.
+ */
+export const getImplementationReport = (
+  allReportsData: Record<string, ReportData>,
+  langImplementation: string,
 ) => {
-  const allImplementations: Record<string, ImplementationDialectCompliance> =
-    {};
-  const dialectCompliance: Record<string, Record<string, Partial<Totals>>> = {};
+  let implementationReport: ImplementationReport | null = null;
+  const dialectCompliance: Record<string, Partial<Totals>> = {};
```

Comment:
How big of a change is it if we make the properties here be `Dialect`s rather than string?

Reply:
Ig its not possible @Julian it shows the below warning

![Screenshot (110)](https://github.com/bowtie-json-schema/bowtie/assets/68469605/a6a78c88-c5e1-47de-8e03-a2770a8c6861)

Reply:
Ig its not possible with `Record` @Julian it shows the below warning

![Screenshot (110)](https://github.com/bowtie-json-schema/bowtie/assets/68469605/a6a78c88-c5e1-47de-8e03-a2770a8c6861)

But it does not throw the warning if we use `Map`. So I'll look into it.

Reply:
@Julian, it worked!, I changed it from `string` to `Dialect` :). Now it looks like this

![Screenshot (111)](https://github.com/bowtie-json-schema/bowtie/assets/68469605/21e52dfc-6690-4b64-acbd-5f2c09e97f75)

Reply:
Great